Userland/Libraries/LibWeb/Layout/InlineLevelIterator.cpp

```diff
@@ -180,14 +180,20 @@ Optional<InlineLevelIterator::Item> InlineLevelIterator::next_without_lookahead(
             m_text_node_context->is_last_chunk = true;
 
         auto& chunk = chunk_opt.value();
-        CSSPixels chunk_width = CSSPixels::nearest_value_for(text_node.font().width(chunk.view) + text_node.font().glyph_spacing());
 
         if (m_text_node_context->do_respect_linebreaks && chunk.has_breaking_newline) {
             return Item {
                 .type = Item::Type::ForcedBreak,
             };
         }
 
+        CSSPixels chunk_width;
+
+        if (m_text_node_context->is_last_chunk)
+            chunk_width = CSSPixels::nearest_value_for(text_node.font().width(chunk.view));
+        else
+            chunk_width = CSSPixels::nearest_value_for(text_node.font().width(chunk.view) + text_node.font().glyph_spacing());
```

Comment:
What is the spec reasoning for this change?

Reply:
As far as I can tell this is LibWeb specific and not covered by spec. In the code, a glyph spacing is added in anticipation for each next chunk, this is so that the calculated text width accounts for two chunks right next to eachother, which would have a glyph spacing inbetween when rendered. I believe it is an oversight that that glyph spacing was being added at the very last chunk, resulting in every TextNode essentially being 1 pixel too wide.

Reply:
If there is a text width calculation spec I'm unaware of I can gladly bring this code up to it